frontend/talentsearch/src/js/components/Router.tsx

```diff
@@ -358,15 +367,15 @@ export const Router: React.FC = () => {
   }
 
   return (
-    <>
+    <Pending fetching={fetching} error={error}>
       <PageContainer
         menuItems={menuItems}
         authLinks={authLinks}
         contentRoutes={[
           ...talentRoutes(talentPaths),
           ...authRoutes(authPaths),
           ...(checkFeatureFlag("FEATURE_APPLICANTPROFILE")
-            ? profileRoutes(profilePaths)
+            ? profileRoutes(profilePaths, data?.me?.id || "")
```

Comment:
> Wouldn't feeding an empty string into the route builder give you garbage routes?
Yes it was.
> Maybe you could add the check that id is not null above with the feature flag check instead of the coalesce?
This would mean none of our profile routes would exist if not logged in which is different than what I think we'd like.

I have fixed this now but I think my solution feels a bit rough.

Reply:
Yeah, the redirect to null isn't great.  What if you broke `profilePaths` into two different collections?  Like
```
        contentRoutes={[
          ...talentRoutes(talentPaths),
          ...authRoutes(authPaths),
          ...(checkFeatureFlag("FEATURE_APPLICANTPROFILE")
            ? anonymousProfileRoutes(profilePaths)
            : []),
          ...((checkFeatureFlag("FEATURE_APPLICANTPROFILE") && data && data.me)
            ? myProfileRoutes(profilePaths, data.me.id)
            : []),
          ...(checkFeatureFlag("FEATURE_DIRECTINTAKE")
            ? directIntakeRoutes(directIntakePaths)
            : []),
        ]}
```

Reply:
This will result in the routes not existing if the user is not signed in versus taking them to sign-in and redirecting them back to this page. I have changed the redirects to fall back to undefined instead of null, I'm not sure if there's really a much better solution. Thoughts?

Reply:
Yeah, I do like that better.  It probably behaves the same way but the code is more expressive of what's going on.  Well done!